src/components/Searchbar.jsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import '../styles/SearchBar.css';
+import SearchIcon from '../assets/searchIcon.svg';
+
+function Searchbar({ onSearch }) {
+  const [searchInput, setSearchInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleSearch = (e) => {
+    const value = e.target.value;
+    setSearchInput(value);
+
+    if (value.length === 0) {
+      setErrorMessage('');
+      if (onSearch) {
+        onSearch('');
+      }
+    } else if (value.length < 3) {
+      setErrorMessage('Search term must be at least 3 characters long');
+    } else {
+      setErrorMessage('');
+      if (onSearch) {
+        onSearch(value);
+      }
+    }
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Enter' && searchInput.length >= 3) {
+      onSearch(searchInput);
+    }
+  };
+
+  return (
+    <div className="search-bar">
+      <input
+        type="text"
```

Comment:
Again we haven't used any common component for such components like input box or dropdown